packages/calypso-e2e/src/lib/pages/cart-checkout-page.ts

```diff
@@ -42,4 +73,105 @@ export class CartCheckoutPage {
 		await this.page.click( selectors.removeCartItemButton( cartItemName ) );
 		await this.page.click( selectors.modalContinueButton );
 	}
+
+	/**
+	 * Obtains the text content of the payment button.
+	 *
+	 * @returns {string} Content of the payment button.
+	 */
+	async getPaymentButtonText(): Promise< string > {
+		const elementHandle = await this.page.waitForSelector( selectors.paymentButton );
+		return await elementHandle.innerText();
+	}
+
+	/**
+	 * Apply a coupon.
+	 *
+	 * If a coupon code is supplied as parameter, this method will apply the supplied code.
+	 * Otherwise, this method will apply the default coupon code found in the configuration.
+	 *
+	 * @param {string} coupon Custom coupon code.
+	 */
+	async enterCouponCode( coupon?: string ): Promise< void > {
+		await this.page.click( selectors.couponCodeInputButton );
+
+		const couponCode =
+			coupon !== undefined ? coupon : ( DataHelper.config.get( 'testCouponCode' ) as string );
+
+		await this.page.fill( selectors.couponCodeInput, couponCode );
+		await this.page.click( selectors.couponCodeApplyButton );
+		await this.page.click( selectors.dismissBanner );
+	}
+
+	/**
+	 * Returns the total amount of items in cart.
+	 *
+	 * @returns {string} Total value of items in cart.
+	 */
+	async getCheckoutTotalAmount(): Promise< number > {
+		const elementHandle = await this.page.waitForSelector(
+			selectors.totalAmount( getTargetDeviceName() )
+		);
+		const stringAmount = await elementHandle.innerText();
+		const parsedAmount = stringAmount.replace( /,/g, '' ).match( /\d+\.?\d*/g );
+		if ( ! parsedAmount?.length ) {
+			throw new Error( 'Unable to locate or parse cart amount.' );
+		}
+		return parseInt( parsedAmount.pop() as string );
```

Comment:
Out of curiosity, why `parseInt` instead of `parseFloat`? With the coupon, you can definitely get cents in American currency at least! And we're matching the decimals in the RegEx!

Reply:
I carried this over from Selenium, where they discarded all decimal points in the comparison.

The idea is that since the test coupon code provides 10% off, the price (36GBP) would change the digit in the ones-place with the coupon code applied.